File: lib/models/source_type.dart
enum SourceType { xtream, m3uUrl, m3u }

String getSourceTypeString(SourceType source) {
  switch (source) {
    case SourceType.m3u:
      return "M3U";
    case SourceType.m3uUrl:
      return "M3U Url";
    case SourceType.xtream:
      return "Xtream";
  }
}

File: rust/src/api/source_type.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[repr(u8)]
pub enum SourceType {
    M3u = 0,
    M3uLink = 1,
    Xtream = 2,
    Custom = 3,
}

impl From<u8> for SourceType {
    fn from(value: u8) -> Self {
        match value {
            0 => SourceType::M3u,
            1 => SourceType::M3uLink,
            2 => SourceType::Xtream,
            3 => SourceType::Custom,
            _ => SourceType::M3u, // Default fallback
        }
    }
}
